src/input.ts

```diff
@@ -210,14 +280,37 @@ export function handleMouseDown(
       if (starting) $head = $anchor;
       else return;
     }
-    const selection = new CellSelection($anchor, $head);
+    const selection = supportRectangularSelection
+      ? getRectangularSelection($anchor, $head)
+      : new CellSelection($anchor, $head);
+
     if (starting || !view.state.selection.eq(selection)) {
       const tr = view.state.tr.setSelection(selection);
       if (starting) tr.setMeta(tableEditingKey, $anchor.pos);
       view.dispatch(tr);
     }
   }
 
+  // get Rectangular cell Selection
+  function getRectangularSelection($anchor: ResolvedPos, $head: ResolvedPos) {
```

Comment:
Let's move this function out of `handleMouseDown` so that we can write test for it.